src/utils.ts

```diff
@@ -96,14 +96,19 @@ export function fallback<T>(...args: T[]): T {
 }
 
 export function extractCredentialsFromUrl(
-	url: string
+	url: string,
+	basePath?: string
 ): Result<{ username: string; password: string; href: string }, "invalid URL"> {
 	try {
 		const { origin, pathname, username, password } = new URL(url);
 		return resultOf({
 			username: decodeURIComponent(username),
 			password: decodeURIComponent(password),
```

Comment:
```suggestion
	const { rtorrentRpcUrl, qbittorrentUrl, transmissionRpcUrl } =
		getRuntimeConfig();
	try {
		const { origin, pathname, username, password } = new URL(url);
		return resultOf({
			username: (rtorrentRpcUrl || qbittorrentUrl || transmissionRpcUrl) ? decodeURIComponent(username) : username,
			password: (rtorrentRpcUrl || qbittorrentUrl || transmissionRpcUrl) ? decodeURIComponent(password) : password,
```

Checks returned getRuntimeConfig() values for which client (once again, deluge not requiring decodeURIComponent() means this should be future-proofed)

Will require adding any other future clients requiring decodeURIComponent to this section, though.

Not sure if there is a more preferred way, this was just something that occurred to me and the first method to accomplish this.

Reply:
we will need it for Deluge

Reply:
The reason we need it for all the other clients isn't due to the way that auth works in those clients, it's due to the way the URL constructor works in Node. 
<img width="622" alt="image" src="https://github.com/cross-seed/cross-seed/assets/9119498/5444095d-b902-4a03-9a87-2947ccc5e705">

So, for the same reasons we need it for other clients, we'll need it for Deluge.

Reply:
I took a further look into what this is actually doing and accomplishing, and was confused - and unaware of ERR_INVALID_URL. 

You're correct. Resolved.